runelite-client/src/main/java/net/runelite/client/plugins/huntertracker/HunterTrackerPlugin.java

```diff
@@ -0,0 +1,208 @@
+
+		pluginToolbar.addNavigation(navButton);
+
+		updatePanel();
+	}
+
+	@Subscribe
+	public void onUsernameChanged(UsernameChanged e)
+	{
+		updatePanel();
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		pluginToolbar.removeNavigation(navButton);
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick t)
+	{
+		if (client.getGameState() != GameState.LOGGED_IN)
+		{
+			// Reset the ticks if you aren't logged in due to VarP initialization
+			tickCount = 0;
+			lastTickLoc = null;
+			return;
+		}
+
+		WorldPoint loc = lastTickLoc;
+		lastTickLoc = client.getLocalPlayer().getWorldLocation();
+		if (loc == null || loc.getRegionID() != lastTickLoc.getRegionID())
+		{
+			return;
+		}
+
+		// It takes a bit for the VarPs to initialize, if you login at the birdhouses give it a few to update
+		if (tickCount <= 10)
+		{
+			tickCount++;
+		}
+
+		BirdhouseRegion region = hunterWorld.getRegions().get(loc.getRegionID());
+		if (region != null && tickCount > 10)
+		{
+			// Write config with new varp and timestamps
+			// huntertracker.<username>.<regionID>.<VarpId>=<varpValue>:<startTime>:<updatedAt>
+			String group = HunterTrackerConfig.KEY_NAME + "." + client.getUsername() + "." + region.getRegionID();
+			long unixNow = Instant.now().getEpochSecond();
+			for (VarPlayer varp : region.getVarps())
+			{
+				// Write the config value if it doesn't match what is current, or it is more than 5 minutes old
+				String key = Integer.toString(varp.getId());
+				String strVarp = Integer.toString(client.getVar(varp));
+				String storedValue = configManager.getConfiguration(group, key);
+				if (storedValue != null)
+				{
+					String[] parts = storedValue.split(":");
+					if (parts.length == 3) {
+						long startTime = 0;
+						long updatedAt = 0;
+						try
+						{
+							startTime = Long.parseLong(parts[1]);
+							updatedAt = Long.parseLong(parts[2]);
+						}
+						catch (NumberFormatException e)
+						{
+							log.debug(e.getMessage());
+						}
+						// If it's been at least 5 minutes and the varp hasn't changed, re-write the config and update
+						if (parts[0].equals(strVarp) && updatedAt + (5 * 60) <= unixNow)
+						{
+							String value = strVarp + ":" + startTime + ':' + unixNow;
+							configManager.setConfiguration(group, key, value);
+							updatePanel();
+						}
+						// If the varp doesn't match, update the config and re-init the startTime
+						else if (!parts[0].equals(strVarp))
+						{
+							String value = strVarp + ":" + unixNow + ":" + unixNow;
+							configManager.setConfiguration(group, key, value);
+							updatePanel();
+						}
+					}
+				}
+				else
+				{
+					// If there's no config value add one
+					String value = strVarp + ":" + unixNow + ":" + unixNow;
+					configManager.setConfiguration(group, key, value);
+					updatePanel();
+				}
+			}
+		}
+	}
+
+	@Schedule(period = 10, unit = ChronoUnit.SECONDS)
+	public void updatePanel()
+	{
+		SwingUtilities.invokeLater(panel::update);
```

Comment:
It may be worth checking whether or not the panel active here to save scheduling this with swing so often.